x/evm/types/msg.go

```diff
@@ -299,6 +299,24 @@ func (msg EthereumTxMsg) Fee() *big.Int {
 	return new(big.Int).Mul(msg.Data.Price, new(big.Int).SetUint64(msg.Data.GasLimit))
 }
 
+// ChainID returns which chain id this transaction was signed for (if at all)
+func (msg *EthereumTxMsg) ChainID() *big.Int {
+	return deriveChainID(msg.Data.V)
+}
+
+// deriveChainID derives the chain id from the given v parameter
```

Comment:
Please provide more context

Reply:
chainID needs to be pulled from an encoded tx [here](https://github.com/ChainSafe/ethermint/pull/87/files#diff-97f02ad1e0a5dd042d5a74fc21e307b9R411). It may be sufficient to just use the chainID from cli context but I added this in the case of a tx submitted with the wrong chainid, the from address will be incorrect.